### ISSUE TITLE
Nested Directory.Build.props not supported

### ISSUE THREAD

**Issue by scabana** [OmniSharp/omnisharp-roslyn#2564]
Hello, 

We have the following setup: 

```xml
<!-- path ./src/Directory.Build.props -->
<Project>
  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />

  <PropertyGroup>
    <!-- Some props to set for projects in this folder -->
  </PropertyGroup>
</Project>

<!-- path ./test/Directory.Build.props -->
<Project>
  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />

  <PropertyGroup>
    <!-- Some props to set for projects in this folder -->
  </PropertyGroup>
</Project>

<!-- path ./Directory.Build.props -->
<Project>
  <PropertyGroup>
    <TargetFramework>net7.0</TargetFramework>
  </PropertyGroup>
</Project>
```

Omnisharp does not seem to understand the `<Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />` and does not end up finding the target framework in the root file. For any project that does not target the latest SDK installed on the machine, VSCode uses the wrong intellisense, syntax and analysers.

Expectation: 
VSCode should support the same experience as MSBuild when it comes to csproj/props setup.

**Comment by JoeRobich** [comment]
@scabana I was unable to reproduce your issue. Could you please set `"omnisharp.loggingLevel": "trace"` in your settings, reopen your project in VS Code, and share the OmniSharp Log from the Output pane? Thanks!

**Comment by scabana** [comment]
humm, interesting, I'm looking into it a little more. I had .net 8 preview installed, uninstalled it and I would still get a `Use primary constructorRoslyn[IDE0290](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0290)` error, which would indicate roslyn targets c# 12. I went to the root `Directory.Build.props` and updated it to have `<LangVersion>11</LangVersion>` and the error did go away. I tried to find logs about which .net version would be loaded, but didn't managed to see anything. So now, I'm running a system with the follwing: 

```
dotnet --list-sdks
6.0.401 [C:\Program Files\dotnet\sdk]
7.0.102 [C:\Program Files\dotnet\sdk]
7.0.305 [C:\Program Files\dotnet\sdk]
7.0.306 [C:\Program Files\dotnet\sdk]
```

I also have a `global.json` with this content: 

```
{
  "sdk": {
    "version": "7.0.102",
    "rollForward": "latestFeature"
  }
}
```

which should also force it to latest .net 7 on my machine.

And it still shows me c# 12 code analysis errors. I tried to clear my whole nuget cache too. I must be missing some cache somewhere that I need to cleanup. 

Thanks a lot for the effort to try and repro!
